src/m365/entra/commands/group/group-user-set.spec.ts

```diff
@@ -0,0 +1,514 @@
+        opts.data.requests[0].method === 'PATCH') {
+        return {
+          responses: Array(2).fill({
+            status: 204,
+            body: {}
+          })
+        };
+      }
+
+      if (opts.url === 'https://graph.microsoft.com/v1.0/$batch' &&
+        opts.data.requests[0].method === 'GET') {
+        return {
+          responses: [
+            {
+              id: userIds[0],
+              status: 200,
+              body: 1
+            },
+            {
+              id: userIds[2],
+              status: 200,
+              body: 1
+            }
+          ]
+        };
+      }
+
+      if (opts.url === 'https://graph.microsoft.com/v1.0/$batch' &&
+        opts.data.requests[0].method === 'DELETE') {
+        return {
+          responses: Array(2).fill({
+            status: 204,
+            body: {}
+          })
+        };
+      }
+
+      throw 'Invalid request';
+    });
+
+    const userNames = userUpns.map(u => ' ' + u).join(',');
+    await command.action(logger, { options: { groupDisplayName: 'Contoso', userNames: userNames, role: 'Owner', verbose: true } });
+    assert.deepStrictEqual(postStub.firstCall.args[0].data.requests, [
+      {
+        id: 1,
+        method: 'PATCH',
+        url: `/groups/${groupId}`,
+        headers: { 'content-type': 'application/json;odata.metadata=none' },
+        body: {
+          'owners@odata.bind': userIds.slice(0, 20).map(u => `https://graph.microsoft.com/v1.0/directoryObjects/${u}`)
+        }
+      },
+      {
+        id: 21,
+        method: 'PATCH',
+        url: `/groups/${groupId}`,
+        headers: { 'content-type': 'application/json;odata.metadata=none' },
+        body: {
+          'owners@odata.bind': userIds.slice(20, 40).map(u => `https://graph.microsoft.com/v1.0/directoryObjects/${u}`)
+        }
+      }
+    ]);
+  });
+
+  it('handles API error when changing roles', async () => {
+    sinon.stub(request, 'post').callsFake(async opts => {
+      if (opts.url === 'https://graph.microsoft.com/v1.0/$batch' &&
+        opts.data.requests[0].method === 'PATCH') {
+        return {
+          responses: [
+            {
+              id: 1,
+              status: 204,
+              body: {}
+            },
+            {
+              id: 2,
+              status: 400,
+              body: {
+                error: {
+                  message: `One or more added object references already exist for the following modified properties: 'members'.`
+                }
+              }
+            }
+          ]
+        };
+      }
+
+      throw 'Invalid request';
+    });
+
+    await assert.rejects(command.action(logger, { options: { groupId: groupId, ids: userIds.join(','), role: 'Member' } }),
+      new CommandError(`One or more added object references already exist for the following modified properties: 'members'.`));
+  });
+
+  it('handles API error when changing roles2', async () => {
```

Comment:
maybe we could provide a better description to this test and also to the next one as well rather than adding indeks. For example we may describe what is the expected outcome or specific error we want to test